code/datums/mind.dm

```diff
@@ -449,6 +464,16 @@
 			return
 		assigned_role = new_job
 
+	if (href_list["greet_role"])
+		var/datum/role/R = locate(href_list["greet_role"])
+		var/chosen_greeting
+		var/custom_greeting
+		if (R.greets.len)
+			chosen_greeting = input("Choose a greeting", "Assigned role", null) as null|anything in R.greets
+			if (chosen_greeting == "custom")
+				custom_greeting = input("Choose a custom greeting", "Assigned role", "")
```

Comment:
Please add `as null||text` to give the user a cancel button. Also you should check if that greeting is null.